python/lsst/ts/cbp/component.py

```diff
@@ -99,26 +99,28 @@ def __init__(self):
         self.mask = None
         self.mask_rotation = None
         self.masks = types.SimpleNamespace(
-            mask1=types.SimpleNamespace(name="Not a mask 1", rotation=0, id=1.),
-            mask2=types.SimpleNamespace(name="Not a mask 2", rotation=0, id=2.),
-            mask3=types.SimpleNamespace(name="Not a mask 3", rotation=0, id=3.),
-            mask4=types.SimpleNamespace(name="Not a mask 4", rotation=0, id=4.),
-            mask5=types.SimpleNamespace(name="Not a mask 5", rotation=0, id=5.),
-            mask9=types.SimpleNamespace(name="Unknown Mask", rotation=0, id=9.))
+            mask1=types.SimpleNamespace(name="Not a mask 1", rotation=0, id=1.0),
+            mask2=types.SimpleNamespace(name="Not a mask 2", rotation=0, id=2.0),
+            mask3=types.SimpleNamespace(name="Not a mask 3", rotation=0, id=3.0),
+            mask4=types.SimpleNamespace(name="Not a mask 4", rotation=0, id=4.0),
+            mask5=types.SimpleNamespace(name="Not a mask 5", rotation=0, id=5.0),
+            mask9=types.SimpleNamespace(name="Unknown Mask", rotation=0, id=9.0),
+        )
         self.mask_dictionary = {
             self.masks.mask1.name: self.masks.mask1,
             self.masks.mask2.name: self.masks.mask2,
             self.masks.mask3.name: self.masks.mask3,
             self.masks.mask4.name: self.masks.mask4,
             self.masks.mask5.name: self.masks.mask5,
-            self.masks.mask9.name: self.masks.mask9}
+            self.masks.mask9.name: self.masks.mask9,
+        }
         self.mask_id_dictionary = {
             self.masks.mask1.id: self.masks.mask1,
             self.masks.mask2.id: self.masks.mask2,
             self.masks.mask3.id: self.masks.mask3,
             self.masks.mask4.id: self.masks.mask4,
             self.masks.mask5.id: self.masks.mask5,
-            self.masks.mask9.id: self.masks.mask9
+            self.masks.mask9.id: self.masks.mask9,
```

Comment:
I strongly encourage you to save `self.mask_dictionary` OR `self.masks` but not both. They contain identical information. As the zen of Python says: "There should be one-- and preferably only one --obvious way to do it." -- in this case, get mask info.

Note that you can get the dict from a SimpleNamespace using __dict__ (which is documented in the example in the Python docs, so it's probably something you can rely on).

Also please reduce the repetition -- it can lead to subtle bugs that are hard to find, due to copy/paste errors. The following few lines do the same thing (though again, I'd only save either mask_diction or self.masks).
```
mask_dict = {f"mask{i}": SimpleNamespace(name=f"Empty {i}", rotation=0, id=i) for i in (1, 2, 3, 4, 5, 6, 9)}
mask_dict["mask9"].name = "Unknown"
self.mask_dictionary = mask_dict
self.masks = SimpleNamespace(**mask_dict)
self.mask_id_dictionary = {value.id: key for key, value in mask_dict.items()}
```
Also I strongly encourage you to write this in a subroutine, so that when mask information is set, all the associated information is set together (self.masks, self.mask_id_dictionary, etc.), which keeps the info in sync. It could take mask names and rotations.